OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs

```diff
@@ -43,13 +43,20 @@ public static IEnumerable<CPos> FrozenUnderFogTiles(Actor a)
 			return Tiles(a.World.Map.Rules, a.Info.Name, a.Info.TraitInfo<BuildingInfo>(), a.Location, true);
 		}
 
-		public static IEnumerable<CPos> UnpathableTiles(string name, BuildingInfo buildingInfo, CPos position)
+		public static IEnumerable<CPos> TargetableTiles(string name, BuildingInfo buildingInfo, CPos position)
 		{
 			var footprint = buildingInfo.Footprint.Where(x => !char.IsWhiteSpace(x)).ToArray();
 			foreach (var tile in TilesWhere(name, buildingInfo.Dimensions, footprint, a => a == 'x'))
 				yield return tile + position;
 		}
 
+		public static IEnumerable<CPos> UnpathableTiles(string name, BuildingInfo buildingInfo, CPos position)
+		{
+			var footprint = buildingInfo.Footprint.Where(x => !char.IsWhiteSpace(x)).ToArray();
```

Comment:
At this stage it would be a lot cleaner to replace all this DIY parsing with a custom `LoadUsing` that stores `buildingInfo.Footprint` as a 2D array of an enum describing these different types.